### PR TITLE
Run analyzer on Travis

### PR DESCRIPTION
**Pull request by goderbauer**
Will require https://github.com/flutter/plugins/pull/79 (analyzer fixes for firebase_database) to be submitted first.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,25 @@
...
+      addons:
+        apt:
+          # Flutter depends on /usr/lib/x86_64-linux-gnu/libstdc++.so.6 version GLIBCXX_3.4.18
+          sources:
+            - ubuntu-toolchain-r-test # if we don't specify this, the libstdc++6 we get is the wrong version
+          packages:
+            - libstdc++6
+            - fonts-droid
+      before_script:
+        - git clone https://github.com/flutter/flutter.git --depth 1
```
Comment (eseidelGoogle) [inline .travis.yml]: I'm surprised travis doesn't have helpers for the git clone step to help reduce work across their fleet.

Comment (goderbauer) [inline .travis.yml]: I think they usually don't expect you to clone a repo in addition to the one under test, which is automatically cloned by Travis for you.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,25 @@
 matrix:
   include:
+    # Job 1) Run analyzer
+    - os: linux
+      jdk: oraclejdk8
```
Comment (mit-mit) [inline .travis.yml]: Why do we need a jdk?

**Comment by mit-mit** [comment]
This is great, thanks!

File: .travis.yml (hunk 2 above)
Comment (goderbauer) [inline .travis.yml]: It's not needed. removed.

**Pull request merged**